src/app/broadcaster.service.ts

```diff
@@ -40,15 +40,11 @@ interface BroadcastEvent {
  */
 @Injectable()
 export class Broadcaster {
-  private _eventBus: Subject<BroadcastEvent>;
-  public static refCount: number = 0;
+  private _eventBus: Subject<BroadcastEvent> = new Subject<BroadcastEvent>();
+  private static instance: Broadcaster;
 
   constructor() {
-    this._eventBus = new Subject<BroadcastEvent>();
-    Broadcaster.refCount++;
-    if (Broadcaster.refCount > 1) {
-      throw new Error('Multiple broadcaster instances detected, this is a fatal error.');
-    }
+    return Broadcaster.instance = Broadcaster.instance || this;
```

Comment:
why aren't you setting the _eventBus anymore?

Reply:
I moved it to the field initialize